File: 01/b/main.py
import sys

x, y = [], []

with open(sys.argv[1]) as f:
    for line in f:
        x_str, y_str = line.strip().split()
        x.append(int(x_str))
        y.append(int(y_str))

print(sum(xx * y.count(xx) for xx in x))

File: 01/b/main.cpp
#include <algorithm>
#include <fstream>
#include <functional>
#include <numeric>
#include <print>
#include <sstream>
#include <string>
#include <vector>

int main(const int, const char** argv) {
  std::vector<int> x, y;
  std::ifstream file{argv[1]};
  std::string line;
  int i, j;

  while (std::getline(file, line)) {
    std::stringstream ss;
    ss << line;
    ss >> i >> j;
    x.push_back(i);
    y.push_back(j);
  }

  auto answer{std::transform_reduce(
    std::cbegin(x),
    std::cend(x),
    0,
    std::plus<>(),
    [&y](const auto n) { return n * std::count(std::cbegin(y), std::cend(y), n); }
  )};

  std::println("{}", answer);
}
